packages/deployer/src/deployer.ts

```diff
@@ -68,11 +68,13 @@ export function deploySolution(
           common.fail(`${model.item.id} is not a Solution Template`)
         );
       } else {
-        // fetch the metadata and pass the item & data forward
+        // fetch the metadata if the model's id is a GUID and pass the item & data forward
         return Promise.all([
           Promise.resolve(model.item),
           Promise.resolve(model.data),
-          common.getItemMetadataAsFile(model.item.id, storageAuthentication)
+          isGuid(model.item.id)
+            ? common.getItemMetadataAsFile(model.item.id, storageAuthentication)
+            : Promise.resolve()
```

Comment:
We're sure that the code further down can handle `itemMetadata` as `undefined`?

Reply:
Yes--`deploySolutionFromTemplate` now checks if the metadata is defined before updating the deployed solution item with it:
```ts
// Pass metadata in via params because item property is serialized, which discards a File
const additionalParams: any = {};
/* istanbul ignore else */
if (solutionTemplateMetadata) {
  additionalParams.metadata = solutionTemplateMetadata;
}
return common.updateItem(
  solutionTemplateBase,
  authentication,
  deployedFolderId,
  additionalParams
);
```